src/common/util/array-move.ts

```diff
@@ -20,20 +20,33 @@ function findNestedItem(
   }, obj);
 }
 
+function updateNestedItem(obj: any, path: ItemPath): any {
+  const lastKey = path.pop()!;
+  const parent = findNestedItem(obj, path);
+  parent[lastKey] = Array.isArray(parent[lastKey])
+    ? [...parent[lastKey]]
+    : [parent[lastKey]];
+  return obj;
+}
+
 export function nestedArrayMove<A>(
   obj: A,
   oldIndex: number,
   newIndex: number,
   oldPath?: ItemPath,
   newPath?: ItemPath
 ): A {
-  const newObj = (Array.isArray(obj) ? [...obj] : { ...obj }) as A;
-  const from = oldPath ? findNestedItem(newObj, oldPath) : newObj;
-  const to = newPath ? findNestedItem(newObj, newPath, true) : newObj;
+  let newObj = (Array.isArray(obj) ? [...obj] : { ...obj }) as A;
 
-  if (!Array.isArray(from) || !Array.isArray(to)) {
-    return obj;
+  if (oldPath) {
+    newObj = updateNestedItem(newObj, [...oldPath]);
   }
```

Comment:
We should keep this.

Reply:
Because "cloneNestedItem" converts to array if it isn't we won't need it.